packages/hub/node-tests/utils/scheduled-payments-test.ts

```diff
@@ -9,80 +9,94 @@ describe('ScheduledPaymentUtils', function () {
-    let now = new Date();
-    expect(() => calculateNextPayAt(now, 0, addMonths(now, 5))).to.throw();
-    expect(() => calculateNextPayAt(now, 32, addMonths(now, 5))).to.throw();
-    expect(() => calculateNextPayAt(now, 1, addMonths(now, 5))).to.not.throw();
-    expect(() => calculateNextPayAt(now, 31, addMonths(now, 5))).to.not.throw();
+    let currentDate = new Date();
+    expect(() => calculateNextPayAt(0, addMonths(currentDate, 5), { currentDate })).to.throw();
+    expect(() => calculateNextPayAt(32, addMonths(currentDate, 5), { currentDate })).to.throw();
+    expect(() => calculateNextPayAt(1, addMonths(currentDate, 5), { currentDate })).to.not.throw();
+    expect(() => calculateNextPayAt(31, addMonths(currentDate, 5), { currentDate })).to.not.throw();
   });
 
   it('sets the last day of month when month has less days than recurringDay case 1', function () {
     // February 2022 has 28 days. If recurringDay is set to 31th, payAt should be set to 28th.
-    let startingDate = new Date(Date.UTC(2022, 1, 1)); // 2022-02-01 (months are zero-indexed)
-    let nextPayAt = calculateNextPayAt(startingDate, 31, addMonths(startingDate, 5));
+    let currentDate = new Date(Date.UTC(2022, 1, 1)); // 2022-02-01 (months are zero-indexed)
+    let nextPayAt = calculateNextPayAt(31, addMonths(currentDate, 5), { currentDate });
 
     expect(nextPayAt?.toISOString()).to.equal('2022-02-28T00:00:00.000Z');
   });
 
   it('sets the last day of month when month has less days than recurringDay case 2', function () {
     // February 2022 has 28 days. If recurringDay is set to 31th, payAt should be set to 28th.
-    let startingDate = new Date(Date.UTC(2022, 1, 27)); // 2022-02-27 (months are zero-indexed)
-    let nextPayAt = calculateNextPayAt(startingDate, 31, addMonths(startingDate, 5));
+    let currentDate = new Date(Date.UTC(2022, 1, 27)); // 2022-02-27 (months are zero-indexed)
+    let nextPayAt = calculateNextPayAt(31, addMonths(currentDate, 5), { currentDate });
 
     expect(nextPayAt?.toISOString()).to.equal('2022-02-28T00:00:00.000Z');
   });
 
   it('sets the last day of month when month has less days than recurringDay case 3', function () {
-    // February 2022 has 28 days. If recurringDay is set to 31st and we're on 28th, next payAt should be set to March 31st
-    let startingDate = new Date(Date.UTC(2022, 1, 28)); // 2022-02-28 (months are zero-indexed)
-    let nextPayAt = calculateNextPayAt(startingDate, 31, addMonths(startingDate, 5));
+    // February 2022 has 28 days. If recurringDay is set to 31st and we're on 28th, next payAt should be set to Feb 28th
+    let currentDate = new Date(Date.UTC(2022, 1, 28)); // 2022-02-28 (months are zero-indexed)
+    let nextPayAt = calculateNextPayAt(31, addMonths(currentDate, 5), { currentDate });
 
-    expect(nextPayAt?.toISOString()).to.equal('2022-03-31T00:00:00.000Z');
+    expect(nextPayAt?.toISOString()).to.equal('2022-02-28T00:00:00.000Z');
```

Comment:
@jurgenwerk Am I correct in this one? Because February is only 28 days so the payment will execute on 28th Feb.

Reply:
Makes sense